pkg/encoding/v1/encoding_test.go

```diff
@@ -105,6 +115,116 @@ func TestPortType_UnmarshalYAML(t *testing.T) {
 	}
 }
 
+func TestPort_UnmarshalYAML(t *testing.T) {
+	tests := []struct {
+		Succeed bool
+		RawPort string
+		Port    Port
+	}{
+		{true, "", Port{}}, // UnmarshalYAML won't be even called for empty strings -> default value
+		{
+			true,
+			`
+port: 5000:80
+host: ""
+`,
+			Port{
+				Port: PortMapping{ContainerPort: 5000, ServicePort: 80},
+				Host: UriAddrFromString(""),
+				Path: UriPathAddrFromString(""), // path defaults to "" when host is validated/set
+			},
+		},
+		{
+			true,
+			`
+port: 5000:80
+host: "subdomain.127.0.0.1.nip.io"
+`,
+			Port{
+				Port: PortMapping{ContainerPort: 5000, ServicePort: 80},
+				Host: UriAddrFromString("subdomain.127.0.0.1.nip.io"),
+				Path: UriPathAddrFromString(""), // path defaults to "" when host is validated/set
+			},
+		},
+		{
+			false, //you have to specify host
+			`
+port: 5000:80
+path: "/admin"
+`,
+			Port{},
+		},
+		{
+			false, //you have to specify host
+			`
+port: 5000:80
+path: ""
+`,
+			Port{},
+		},
+		{
+			true,
+			`
+port: 5000:80
+host: ""
+path: ""
+`,
+			Port{
+				Port: PortMapping{ContainerPort: 5000, ServicePort: 80},
+				Host: UriAddrFromString(""),
+				Path: UriPathAddrFromString(""),
+			},
+		},
+		{
+			true,
+			`
+port: 5000:80
+host: ""
+path: "/admin"
+`,
+			Port{
+				Port: PortMapping{ContainerPort: 5000, ServicePort: 80},
+				Host: UriAddrFromString(""),
+				Path: UriPathAddrFromString("/admin"),
+			},
+		},
+		{
+			true,
+			`
+port: 5000:80
+host: "subdomain.127.0.0.1.nip.io"
+path: "/admin"
+`,
+			Port{
+				Port: PortMapping{ContainerPort: 5000, ServicePort: 80},
+				Host: UriAddrFromString("subdomain.127.0.0.1.nip.io"),
+				Path: UriPathAddrFromString("/admin"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run("", func(t *testing.T) {
```

Comment:
I don't want to. Naming just for the sake of naming is wrong. Naming is hard and I would rather have unnamed test than those with misleading descriptions. Especially with the feature that if you specify empty string there and it fails, it will tell you the exact test index. good enough for me.

This feature is for generic naming when you can actually substitute the test case into some formatted string like `fmt.Sprintf("%s in %s", tc.gmt, tc.loc)` - not for structures taking 20 lines to initialize.